### PR TITLE
Fix issues related to multicast UDP subnet checks.

### PR DESCRIPTION
**Pull request by abhijitherekar**
## Description

More fixes made to the multicast UDP range checks.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by philipatl — COMMENTED** [review]
Just to be complete, I think there's one more case this doesn't handle.
`"0.0.0.0/0", "!224.0.0.0/5", "!232.0.0.0/5"`
This is equivalent to
`"0.0.0.0/0", "!224.0.0.0/4"`
Is it possible to check this with the cidranger code?

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "udp-multi",
    "component": "gaia",
    "pr-id": "617",
    "commit-sha": "f162801675e398bcab3235e99d3c9708dfb13efd",
    "pipeline": "master"
  },
  {
    "project": "udp-multi",
    "component": "backend",
    "pr-id": "1056",
    "commit-sha": "9e362813c31b0d1892a9ac253b9d6db65f910795",
    "pipeline": "master"
  }
]
```

**Review by philipatl — DISMISSED** [review]

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "udp-multi",
    "component": "backend",
    "pr-id": "1056",
    "commit-sha": "9e362813c31b0d1892a9ac253b9d6db65f910795",
    "pipeline": "master"
  },
  {
    "project": "udp-multi",
    "component": "gaia",
    "pr-id": "617",
    "commit-sha": "04dfc7c0f7f892be24b640f140fb1fd32cb8af37",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "udp-multi",
    "component": "gaia",
    "pr-id": "617",
    "commit-sha": "04dfc7c0f7f892be24b640f140fb1fd32cb8af37",
    "pipeline": "master"
  },
  {
    "project": "udp-multi",
    "component": "backend",
    "pr-id": "1056",
    "commit-sha": "9e362813c31b0d1892a9ac253b9d6db65f910795",
    "pipeline": "master"
  }
]
```

File: netutils/cidr_test.go [hunk 1]
```diff
@@ -391,6 +391,83 @@ func Test_ValidateUDPCIDRs(t *testing.T) {
...
+		{
+			name: "all multicast subnets excluded test and everything under 128/1 included",
+			args: args{
+				[]string{"0.0.0.0/0", "128.0.0.0/1", "!224.0.0.0/5", "!224.0.0.0/4", "!228.10.10.10/32", "240.0.0.0/8"},
+			},
+			wantErr: false,
+		},
+		{
+			name: "multicast 224/5, 232/5 excluded but 224/4(232.x.x.x/4) and 128/1 included",
+			args: args{
```
Comment (philipatl) [inline netutils/cidr_test.go:424]: If you're in this code again, maybe you could do a similar IPv6 test (where `multicastv6Subnet` is not explicitly excluded, but the union of smaller CIDRs compose to exclude it) to exercise `addressv6RangeCount` some more, but I don't think it's a big deal right now.

**Review by philipatl — APPROVED** [review]

File: netutils/cidr_test.go (hunk 1 above)
Comment (abhijitherekar) [reply netutils/cidr_test.go:424]: oh ya.. I should have done it. my bad, yes if I revisit I will sure add them.

**Pull request merged**